lib/core/backend/backendSubscription.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2022 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JSONObject } from "kuzzle-sdk";
+import { Connection, KuzzleRequest } from "../../api/request";
+import * as kerror from "../../kerror";
+import { ApplicationManager } from "./index";
+
+const runtimeError = kerror.wrap("plugin", "runtime");
+
+export class BackendSubscription extends ApplicationManager {
+  /**
+   * Registers a new realtime subscription on the specified connection
+   *
+   * @param connection Connection to register the subscription on
+   * @param index Index name
+   * @param collection Collection name
+   * @param filters Subscription filters
+   * @param options.volatile Volatile data
+   * @param options.scope Subscription scope
+   * @param options.users Option for users notifications
+   */
+  async add(
```

Comment:
I can't remember if this could be a problem but what about the `sdk.as` function of the backend SDK instance? Should this function support it?

Reply:
I don't see any usage of impersonation with the realtime engine since we don't change document metadata. Are you thinking about something else?

Reply:
No I'm not. Just wanted to make you think about it just in case 🙂

Reply:
Ok, I don't see any then :+1: